### DISCUSSION TITLE
Netbox IP assignment to Windows VM not working

### DISCUSSION THREAD

**Discussion by ScorpionKing34** [netbox-community/netbox#14034] [Help Wanted!]
No questions any more

**Comment by gdprdatasubect** [comment]
Use config contexts?

**Comment by ScorpionKing34** [comment]
dear gdprdatasubect,

I at the moment not at work where the test  Netbox ansible lab setup is configured and I am the next week free. So I cannot share some contexts about the configs. But is there a tutorial or general steps to succesfully deploy Windows VM's with Netbox and given IP/Sub/DNS/Gateway?

I couldn't find it so quickly.

**Reply by gdprdatasubect** [reply]
https://docs.netbox.dev/en/stable/features/context-data/ you should use this feature with ansible, pay close attention to "Hierarchical Rendering"

**Comment by candlerb** [comment]
Your question is garbled.

It sounds like you are asking "how can I use Ansible to configure the ethernet interface and gateway on a newly-created Windows Server VM?"  If so, that's a question about Ansible, not Netbox.  Use a static inventory instead of Netbox-ansible to simplify the question, then ask it again on an Ansible mailing list.

In any case, Ansible needs a way to [talk to the Windows server](https://docs.ansible.com/ansible/latest/os_guide/index.html), so you have a chicken-and-egg situation: how can you use Ansible to configure the IP address, if you can't already communicate with it over IP? Maybe Ansible has some out-of-band way to communicate with Windows servers. But if so, that's also a question about Ansible, and nothing to do with Netbox.

Netbox-ansible is purely an inventory source for Ansible: i.e. providing the list of machines to manage from Netbox's database. Neither Netbox nor Netbox-ansible performs any actual configuration.

For a newly-created Linux VM, you'd typically use cloud-init to configure the IP address and gateway at the time the VM first boots, as well as setting credentials, so that Ansible (or whatever) could talk to it once it's running. I don't know if Windows VMs created under VMware have a similar initialization mechanism, but again, that's not a question about Netbox.

**Comment by ScorpionKing34** [comment]


**Reply by candlerb** [reply]
> What I can remember for the setup at work Lab is that when I created a Windows VM, I saw in some inventory file the created IP in Netbox web interface that is linked to a Windows virtual machine.

"Some inventory file" - where exactly did you see this? And what integration have you built, so that creation of a VM creates something in the Netbox database?

If somebody else built this lab environment for you, then they'll understand what they built and they'll be able to explain it to you. Asking us isn't much use, since we've no idea how your environment was built. Netbox is just a database, around which other integrations can be built.

> a answer file containing a static IP.

From what you describe now, it seems you are creating the VM and telling it what IP address and gateway to use at creation time in "an answer file". That is different to the question you originally asked, where you said: "the configured IP address in the interface 'Ethernet0' that is linked to the VM in Netbox does not configured in the Windows server 2016 VM"

Maybe you could create this "answer file" from data in Netbox.

> Side question: Does Netbox has also fields for entering DNS and gateway addressess?

In Netbox you would typically use Config Contexts for DNS server configuration: it lets you set a default for a group of machines (e.g. all machines in a particular Site), and override it, e.g. at the level of Rack or individual device.

There is no built-in way to model a "Gateway" in Netbox, and there are various options. You could record this as a custom field on a Prefix, or you could have an individual IP address object with a tag saying it's a gateway. It depends on how you want to consume this information: for example, whether you are using Netbox as a source of truth to *push* configs to devices.

> But for now I need to get the IP address manually that the VM from a DHCP server gets.

Now you're saying something else again: that the Windows VM picks up its IP address via DHCP when it starts.  Then you'll need to look at your DHCP server logs to find the IP address, or have some sort of agent which registers the newly-created VM (which could include creating it in Netbox), or you might be able to use the VMware API to scrape the VM MAC and/or IP addresses.

Again, this depends on your processes: do you want to use Netbox as a source-of-truth, so that creating a VM in Netbox triggers the VMware API to spin up the VM? Or do you want to spin up the VM in VMware, and retrospectively document the VM in Netbox? Either way, you have to build this integration yourself.

I'm going to drop out of this conversation now, as I don't think you've explained clearly what you've already built, let alone how Netbox fits into this.

**Comment by ScorpionKing34** [comment]
Sorry for the confusion.  I'll will remove all of my questions.
